### PR TITLE
Ben fuel pickup functionality

### PR DESCRIPTION
**Pull request by ballardi**
- fuelpickup prefab (tested with multiple in scene)
- ropesource that lowers/raises ropehook. also tells hook when to start grabbing stuff (based on configurable distance from ropesource). also when hook is raised back to ropesource if it's grabbing something that something will be removed and the hook will be reset.
- ropehook that grabs items it collides with.

outstanding features not yet done: https://trello.com/c/LdA9DS0u/82-fuel-pickup-v2

... also moved a bunch of assets and dirs around for organization's sake. regret doing that in the same MR now..

### CODE REVIEW

File: Assets/Player/GunshipRoot.prefab [hunk 1]
```diff
@@ -95,6 +95,7 @@ Transform:
   - {fileID: 1761745919798031696}
   - {fileID: 1761745918375830818}
   - {fileID: 1519918971610312065}
+  - {fileID: 6612633233016530924}
   m_Father: {fileID: 5591879906522224616}
```
Comment (p3zd1sp3nx0r) [inline Assets/Player/GunshipRoot.prefab:99]: It sucks reviewing these prefabs....

File: Assets/Player/MovementScript.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    // The force applied per unit of distance below the desired height.
+    float hoverForce = 5.0f;
+
+    // The amount that the lifting force is reduced per unit of upward speed.
+    // This damping tends to stop the object from bouncing after passing over
+    // something.
+    float hoverDamp = 0.5f;
+
+    Rigidbody _rigidBody;
```
Comment (p3zd1sp3nx0r) [inline Assets/Player/MovementScript.cs:25]: Are we using underscores for private variables?

File: Assets/Player/MovementScript.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+            QueryTriggerInteraction.Collide); // raycasts should ignore colliders marked as triggers
+
+        // go through everything we hit, and return distance to terrain hit
+        for (int i = 0; i < raycastHits.Length; i++) {
+            if(raycastHits[i].transform.GetComponent<TerrainCollider>() != null) {
+                return raycastHits[i].distance;
+            }
+        }
+
+        throw new System.Exception("should never happen: if we can't find terrain under us, we have a big problem. hits:"+raycastHits.Length);
```
Comment (p3zd1sp3nx0r) [inline Assets/Player/MovementScript.cs:82]: Is this the standard way we want to handle exceptions?

File: Assets/Player/RopeSource/RopeHookScript.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
+﻿using UnityEngine;
+
+public class RopeHookScript : MonoBehaviour {
+
+    // set in editor
+    public bool DebugMode;
```
Comment (p3zd1sp3nx0r) [inline Assets/Player/RopeSource/RopeHookScript.cs:6]: Nice, I gotta remember that

File: Assets/Player/MovementScript.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+{
+    // Forward movement force.
+    public float moveForce = 20.0f;
+
+    // Torque for left/right rotation.
+    public float rotateTorque = 5.0f;
+
+    // Desired hovering height.
+    public float hoverHeight = 25.0f;
+
```
Comment (p3zd1sp3nx0r) [inline Assets/Player/MovementScript.cs:16]: No tooltips? Room to at more parameters too

File: Assets/Player/RopeSource/RopeSourceScript.cs [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+
+    public void OnTriggerExit(Collider other) {
+        if (DebugMode) {
+            Debug.Log("OnTriggerExit "+other.gameObject.name);
+        }
+        if (other.tag == "Item") {
+            StartRaising();
+        }
+    }
+
```
Comment (p3zd1sp3nx0r) [inline Assets/Player/RopeSource/RopeSourceScript.cs:116]: Lots of Methods, I like :)

**Review by p3zd1sp3nx0r — APPROVED** [review]

**Comment by p3zd1sp3nx0r** [comment]
Merged

File: Assets/Player/MovementScript.cs (hunk 3 above)
Comment (ballardi) [reply Assets/Player/MovementScript.cs:82]: in production there's an argument to ignore unexpected behavior to potentially avoid the user realizing, but in our case i'd rather know immediately if something that should never happen has happened. i guess if we were serous we could have an if(env!=prod) throw.. sorta thing but that'd be easy to add later (just find all cases we throw.

File: Assets/Player/MovementScript.cs (hunk 5 above)
Comment (ballardi) [reply Assets/Player/MovementScript.cs:16]: yeah these were just moves of what cotter initially wrote. i didn't spend time trying to understand them cos it didn't impact what i was in there to change, so wasn't able to write the tooltips

File: Assets/Player/RopeSource/RopeHookScript.cs [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        this.GetComponent<BoxCollider>().enabled = NewEnabledState;
+    }
+
+    public void OnTriggerEnter(Collider other) {
+
+        if (other.tag != "Item") {
+            return;
+        }
+
+        if (DebugMode) {
```
[the remark this answers is not in the document]
Comment (ballardi) [reply Assets/Player/RopeSource/RopeHookScript.cs:49]: how else do we do it? you mean sometimes we just do Debug.log without an if? IIRC the console is usually very empty so i assumed we were wrapping. maybe we just aren't using debug.log much?

**Pull request merged**